frames/src/main/java/com/checkout/frames/component/expirydate/ExpiryDateViewModel.kt

```diff
@@ -88,8 +88,12 @@ internal class ExpiryDateViewModel @Inject constructor(
             componentState.expiryDate.value = result.value
             componentState.hideError()
             paymentStateManager.expiryDate.value = result.value
-            paymentStateManager.isExpiryDateValid.update { true }
             wasInvalidPastExpiryDate = false
+            if (!isFocused || componentState.expiryDate.value.length == componentState.expiryDateMaxLength.value) {
```

Comment:
Curious why we need to check `!isFocused` before updating `isExpiryDateValid`?

Also why we need another length check if the validation result is Success?